integration-tests/test/errors.js

```diff
@@ -0,0 +1,80 @@
+import {blockedClassError, compilationError, theaterRuntimeFileNotFound, blankProject} from "./lib/sources.js";
+import {CONSOLE, THEATER} from "./lib/MiniAppType.js";
+import {
+  assertMessagesEqual,
+  verifyMessages,
+  COMPILING_STATUS_MESSAGE,
+  INITIAL_STATUS_MESSAGES,
+  EXIT_STATUS_MESSAGE,
+} from "./helpers/testHelpers.js";
+import {expect} from "chai";
+import {uploadSources} from "./lib/JavabuilderConnectionHelper.js";
+
+describe("Errors", () => {
+  it("Compilation Error", (done) => {
+    const expectedMessages = [
+      COMPILING_STATUS_MESSAGE,
+      { type: "SYSTEM_OUT", value: "/HelloWorld.java:1: error: reached end of file while parsing\npublic class HelloWorld {\n                         ^\n" },
+      { type: "EXCEPTION", value: "COMPILER_ERROR" },
+      EXIT_STATUS_MESSAGE
+    ];
+
+    const assertOnMessagesReceived = receivedMessages => assertMessagesEqual(receivedMessages, expectedMessages);
+    verifyMessages(compilationError, CONSOLE, assertOnMessagesReceived, done);
+  }).timeout(20000);
+
+  it("Runtime Error", (done) => {
+    const expectedMessages = [
+      ...INITIAL_STATUS_MESSAGES,
+      {
+        type: "EXCEPTION",
+        value: "FILE_NOT_FOUND",
+        detail: {
+          causeMessage: "dog.jpeg",
+          cause: "Exception message: java.io.FileNotFoundException: dog.jpeg",
+          fallbackMessage: "Exception message: java.io.FileNotFoundException: dog.jpeg"
+        }
+      },
+      EXIT_STATUS_MESSAGE
+    ];
+
+    const assertOnMessagesReceived = receivedMessages => assertMessagesEqual(receivedMessages, expectedMessages, validateExceptionDetailKey);
+    verifyMessages(theaterRuntimeFileNotFound, THEATER, assertOnMessagesReceived, done);
+  }).timeout(20000);
+
+  it("Uses blocked classes", (done) => {
+    const expectedMessages = [
+      ...INITIAL_STATUS_MESSAGES,
+      {
+        type: "EXCEPTION",
+        value: "INVALID_CLASS",
+        detail: {
+          causeMessage: "java.lang.Thread",
+          cause: "Exception message: java.lang.ClassNotFoundException: java.lang.Thread",
+          fallbackMessage: "Exception message: java.lang.ClassNotFoundException: java.lang.Thread"
+        }
+      },
+      EXIT_STATUS_MESSAGE
+    ];
+
+    const assertOnMessagesReceived = receivedMessages => assertMessagesEqual(receivedMessages, expectedMessages, validateExceptionDetailKey);
+    verifyMessages(blockedClassError, CONSOLE, assertOnMessagesReceived, done);
+  }).timeout(20000);
+
+  it("Does not have valid token", async () => {
+    // We don't appear to get any more detail than a 500 response when an invalid token is provided.
+    const httpResponse = await uploadSources(blankProject, 'aBadToken');
+    expect(httpResponse.status).to.equal(500);
```

Comment:
This could be a follow-up, but I believe we should get a 4xx error code if the payload is signed with the wrong key (but still well-formed) if you want to add a test for that? I think the 500 is coming from the authorizer throwing an error rather than the authorizer rejecting the token

Reply:
oh I see, like the key needs to be the right length and we'll get 4xx...I can update

Reply:
Are there any best practices for stubbing this sort of stuff? Different encryption types are a bit over my head -- do you know where I can see the configuration we use to generate real javabuilder keys? I see something like this, but it requires a bunch of options I don't know anything about:

https://nodejs.org/api/crypto.html#cryptogeneratekeypairtype-options-callback

Alternatively, are there any best practices for stubbing this sort of stuff?

Reply:
I renamed the test to "malformed token", so I might just take the easy way out and merge this version

Reply:
Yeah that's totally fair :) I think it'll be little involved to generate a valid but incorrect key. Maybe we can take a look it at it when we add throttling error tests